lib/generators/nav-generator.rb

```diff
@@ -21,16 +25,27 @@ def nav_for_parent(page_list, parentPage, grandParentPage)
         # Attach the child data to the page itself, used to render the footer Table of Contents
         page.data['children'] = children
         
-        # Attach parent/grandparent URLs for rendering breadcrumb
-        page.data['parent_url'] = parentPage&.url
-        page.data['grand_parent_url'] = grandParentPage&.url
-        
         {
           'title' => page.data['title'],
           'url' => page.url,
           'children' => children
         }
       }
     end
+
+    def breadcrumb_for_parent(page_list, parentPage, grandParentPage)
+      parentPageTitle = parentPage && parentPage.data['title']
+      pages = page_list
+        .filter { |page| page.data['parent'] == parentPageTitle && page.data['title'] != nil }
+        .sort_by { |page| [page.data['nav_order'] || 999, page.data['title']] }  
+
+      nav = pages.map { |page|
+        children = breadcrumb_for_parent(page_list, page, parentPage)
+        
+        # Attach parent/grandparent URLs for rendering breadcrumb
+        page.data['parent_url'] = parentPage&.url
+        page.data['grand_parent_url'] = grandParentPage&.url
```

Comment:
Might want to specify the return here just to be sure what ruby is aggregating

```suggestion
        page.data['grand_parent_url'] = grandParentPage&.url
        
        {
          'parent_url' = parentPage&.url
          'grand_parent_url' = grandParentPage&.url
        }
```